File: holoken-core/src/main/java/com/holokenmod/options/GameDifficulty.java
package com.holokenmod.options;

public enum GameDifficulty {
	VERY_EASY(0.0),
	EASY(5.0),
	MEDIUM(33.33),
	HARD(66.67),
	EXTREME(95.0);
	
	private final double minimumPercentage;
	
	GameDifficulty(double minimumPercentage) {
		this.minimumPercentage = minimumPercentage;
	}
}

File: gauguin-core/src/main/kotlin/org/piepmeyer/gauguin/difficulty/GameDifficulty.kt
package org.piepmeyer.gauguin.difficulty

enum class GameDifficulty {
    VERY_EASY,
    EASY,
    MEDIUM,
    HARD,
    EXTREME, ;

    companion object {
        fun all(): Set<GameDifficulty> = entries.toSet()
    }
}
